src/ripple/app/misc/AmendmentTableImpl.cpp

```diff
@@ -97,51 +101,110 @@ class AmendmentTableImpl : public AmendmentTable
     amendmentList_t getDesired();    // amendments we support, do not veto, are not enabled
 };
 
-void
-AmendmentTableImpl::addInitial ()
+namespace detail
 {
-    // For each amendment this version supports, construct the AmendmentState object by calling
-    // addKnown. Set any vetoes or defaults. A pointer to the AmendmentState can be stashed
+using PreEnabledAmendmentsCollection = std::vector<AmendmentName>;
+
+/** preEnabledAmendments is a static collection of amendments are are enabled at build time.
+
+   Add amendments to this collection at build time to enable them on this server.
+*/
+
+PreEnabledAmendmentsCollection const preEnabledAmendments;
 }
 
-AmendmentState*
-AmendmentTableImpl::getCreate (uint256 const& amendmentHash, bool create)
+void
+AmendmentTableImpl::addInitial (Section const& section)
 {
-    // call with the mutex held
-    auto iter (m_amendmentMap.find (amendmentHash));
-
-    if (iter == m_amendmentMap.end())
+    for (auto const& a : detail::preEnabledAmendments)
     {
-        if (!create)
-            return nullptr;
+        if (!a.valid ())
+        {
+            std::string const errorMsg =
+                (boost::format (
+                     "preEnabledAmendments contains an invalid hash (expected "
+                     "a hex number). Value was: %1%") %
+                 a.hexString ()).str ();
+            throw std::runtime_error (errorMsg);
```

Comment:
Thoughts for the future.  If there's an error in `preEnabledAmendments` that's a programming error, right?  So you might consider an additional `assert` here so we get stopped dead in debug mode.  You have to leave the `throw`, because the `assert` will get compiled out in release mode.

Also, for my own education, is `boost::format` an improvement over what is, basically, string concatenation?